### PR TITLE
add close to watch

### LINKED ISSUE #94: Leak from OkHttpClient
on performing repeated watches on custom objects I eventually see in logs

```2017-10-14 15:26:12.773  WARN 5 --- [ ConnectionPool] com.squareup.okhttp.OkHttpClient         : A connection to https://10.0.0.1/ was leaked. Did you forget to close a response body?```

I don't see any way to close the response body from the Watch wrappers or gain access to it to close manually.

I'm doing:
```java
CustomObjectsApi api = new CustomObjectsApi();
		Watch<Object> watch = Watch.createWatch(
                client,
                api.listNamespacedCustomObjectCall("xxx.yyy", "v1alpha1", "default", "mykind", null, null, rs, true, null, null),
                new TypeToken<Watch.Response<Object>>(){}.getType());
```

### PR DESCRIPTION
**Pull request by cliveseldon**
fixes #94

### CODE REVIEW

File: util/src/main/java/io/kubernetes/client/util/Watch.java [hunk 1]
```diff
@@ -120,4 +120,8 @@ public boolean hasNext() {
     public void remove() {
         throw new UnsupportedOperationException("remove");
     }
+    
+    public void close() throws IOException {
```
Comment (lwander) [inline util/src/main/java/io/kubernetes/client/util/Watch.java]: I'm wondering if this is meant to go in `remove()` - @brendandburns?

Comment (brendandburns) [reply util/src/main/java/io/kubernetes/client/util/Watch.java]: No, `remove()` is part of implementing the Java `Iterable` interface, (and it's legal for it to throw if you don't support it...)

We should mark `remove()` with `@Override` to make this clear.

**Comment by brendandburns** [comment]
Generally LGTM.  You need to sign the CLA.

One comment: can you make the class implement `java.lang.Closeable` since we're adding that method.

Thanks!
--brendan

**Comment by cliveseldon** [comment]
I signed a CLA

**Comment by brendandburns** [comment]
LGTM, thanks!

**Pull request merged**